### PR TITLE
Increase consumer confidence + adapt to 24.0

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
For reference...

* OTP 23

```erlang
1> compile:file("_build/test/test_gen/duplicates.erl", [{i, "include"}, return_errors]).
_build/test/test_gen/duplicates.erl:5 Unable to resolve imported module bar
{error,[{"_build/test/test_gen/duplicates.erl",
         [{none,compile,
                {parse_transform,mixer,{error,undef_mixin_module}}}]}],
       []}
```

* OTP 24

```erlang 
1> compile:file("_build/test/test_gen/duplicates.erl", [{i, "include"}, return_errors]).
_build/test/test_gen/duplicates.erl:{5,2} Unable to resolve imported module bar

*** Internal compiler error ***
exception exit: {error,undef_mixin_module}
  in function  mixer:expand_mixin/2 (/Users/paulo.oliveira/Development/contrib/mixer/src/mixer.erl, line 140)
  in call from mixer:'-parse_and_expand_mixins/2-lc$^0/1-0-'/2 (/Users/paulo.oliveira/Development/contrib/mixer/src/mixer.erl, line 114)
  in call from mixer:'-parse_and_expand_mixins/2-lc$^0/1-0-'/2 (/Users/paulo.oliveira/Development/contrib/mixer/src/mixer.erl, line 114)
  in call from mixer:parse_and_expand_mixins/2 (/Users/paulo.oliveira/Development/contrib/mixer/src/mixer.erl, line 114)
  in call from mixer:parse_transform/2 (/Users/paulo.oliveira/Development/contrib/mixer/src/mixer.erl, line 42)
  in call from compile:foldl_transform/3 (compile.erl, line 1145)
  in call from compile:fold_comp/4 (compile.erl, line 416) 
  in call from compile:internal_comp/5 (compile.erl, line 400)
error
```

**Comment by paulo-ferraz-oliveira** [comment]
Potentially related "commit": https://github.com/erlang/otp/commit/746fb27c38c7e18dfad7ca540b8716c5821da587#diff-3f0859234ee3b499e4eeec6cf88815b8897badc37473a34d2a47f6d13e768ef6

**Comment by paulo-ferraz-oliveira** [comment]
Still unsure if there's a proper fix for this, or if throwing from a `parse_transform` is Okay, which would mean OTP 24 needs a tweak: https://github.com/erlang/otp/commit/746fb27c38c7e18dfad7ca540b8716c5821da587#r51045733.

**Comment by paulo-ferraz-oliveira** [comment]
It seems that where the previous version of the compiler was more relaxed with certain exceptions, now the `error:` class is expected. I'm pushing a `mixer` fix that hopefully will solve problems in OTP 24 while working for OTP 23 and older siblings.

File: src/mixer.erl [hunk 1]
```diff
@@ -137,7 +137,7 @@ expand_mixin(Line, Name) when is_atom(Name) ->
             io:format(
                 "~s:~p Unable to resolve imported module ~p~n",
                 [get_file_name(), Line, Name]),
-            exit({error, undef_mixin_module});
+            error({error, undef_mixin_module});
```
Comment (paulo-ferraz-oliveira) [inline src/mixer.erl:140]: @elbrujohalcon, let me know if this is acceptable. I left a few notes on the Conversation on why I think this is a proper fix.

Comment (elbrujohalcon) [reply src/mixer.erl:140]: Yeah! It's cool.

**Review by elbrujohalcon — APPROVED** [review]

**Pull request merged**